rskj-core/src/test/java/co/rsk/peg/bitcoin/BitcoinUtilsTest.java

```diff
@@ -224,6 +224,40 @@ void getFirstInputSigHash_forP2shP2wshTx_shouldReturnExpectedLegacyTxSigHash() {
         assertEquals(expectedSigHash, sigHash.get());
     }
 
+    @Test
+    void extractRedeemScriptFromInput_forP2shP2wshTx_withDifferentRedeemScriptInputs_shouldExtractThemProperly() {
+        // Arrange
+        BtcTransaction btcTx = new BtcTransaction(btcMainnetParams);
+
+        int outputIndex = 0;
+        int nHash = 0;
+        Federation federation = P2shErpFederationBuilder.builder().build();
+        Script redeemScript = federation.getRedeemScript();
+        Script p2shMultiSigScriptSig = federation.getP2SHScript().createEmptyInputScript(null, redeemScript);
+        btcTx.addInput(BitcoinTestUtils.createHash(nHash), outputIndex, p2shMultiSigScriptSig);
+
+        nHash++;
+        Script emptyScript = new Script(new byte[]{});
+        BtcECKey pubKey = BitcoinTestUtils.getBtcEcKeyFromSeed("abc");
+        Script anotherRedeemScript = ScriptBuilder.createRedeemScript(1, List.of(pubKey));
```

Comment:
```suggestion
        Federation segwitCompatibleFederation = P2shP2wshErpFederationBuilder.builder().build();
        Script redeemScript2 = segwitCompatibleFederation.getRedeemScript();
```

no big deal, just a suggestion

Reply:
I didn't know there was a P2shP2wshErpFederationBuilder :D now added